### ISSUE TITLE
Drives missing although everything seems to work

### ISSUE THREAD

**Issue by PieterAppronto** [adriankumpf/teslamate#2078]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

I don't see any of me drives after a certain date, however Teslamate seems to be fully functional. I can always see current location and state etc. and Grafana works.

I will export a part of the log, because the command for the log has been running now for 10 minutes and is still not finished.

### Expected Behavior

I would expect to see a drive I make show up in Grafana

### Steps To Reproduce

1. Go to Drives dashboard.
2. I dont see my current drives

### Relevant log output

```shell
teslamate_1  |             (stdlib 3.16) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  |     (elixir 1.12.3) lib/gen_server.ex:1024: GenServer.call/3
teslamate_1  |     (elixir 1.12.3) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
teslamate_1  |     (elixir 1.12.3) lib/task/supervised.ex:35: Task.Supervised.reply/5
teslamate_1  |     (stdlib 3.16) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  | Function: &:erlang.apply/2
teslamate_1  |     Args: [#Function<4.48828428/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [doors_open: false]]
teslamate_1  | 2021-10-06 17:35:14.310 [error] Task #PID<0.16520.0> started from #PID<0.16491.0> terminating
teslamate_1  | ** (stop) exited in: GenServer.call(TeslaMate.Mqtt.Publisher, {:publish, "teslamate/cars/1/est_battery_range_km", "135.72", [retain: true, qos: 1]}, 10000)
teslamate_1  |     ** (EXIT) an exception was raised:
teslamate_1  |         ** (WithClauseError) no with clause matching: {:error, :timeout}
teslamate_1  |             (tortoise 0.10.0) lib/tortoise.ex:269: Tortoise.publish/4
teslamate_1  |             (teslamate 1.24.2) lib/teslamate/mqtt/publisher.ex:41: TeslaMate.Mqtt.Publisher.handle_call/3
teslamate_1  |             (stdlib 3.16) gen_server.erl:721: :gen_server.try_handle_call/4
teslamate_1  |             (stdlib 3.16) gen_server.erl:750: :gen_server.handle_msg/6
teslamate_1  |             (stdlib 3.16) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  |     (elixir 1.12.3) lib/gen_server.ex:1024: GenServer.call/3
teslamate_1  |     (elixir 1.12.3) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
teslamate_1  |     (elixir 1.12.3) lib/task/supervised.ex:35: Task.Supervised.reply/5
teslamate_1  |     (stdlib 3.16) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  | Function: &:erlang.apply/2
teslamate_1  |     Args: [#Function<4.48828428/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [est_battery_range_km: 135.72]]
teslamate_1  | 2021-10-06 17:35:14.314 [error] Task #PID<0.16514.0> started from #PID<0.16491.0> terminating
teslamate_1  | ** (stop) exited in: GenServer.call(TeslaMate.Mqtt.Publisher, {:publish, "teslamate/cars/1/power", "-11", [retain: true, qos: 1]}, 10000)
teslamate_1  |     ** (EXIT) an exception was raised:
teslamate_1  |         ** (WithClauseError) no with clause matching: {:error, :timeout}
teslamate_1  |             (tortoise 0.10.0) lib/tortoise.ex:269: Tortoise.publish/4
teslamate_1  |             (teslamate 1.24.2) lib/teslamate/mqtt/publisher.ex:41: TeslaMate.Mqtt.Publisher.handle_call/3
teslamate_1  |             (stdlib 3.16) gen_server.erl:721: :gen_server.try_handle_call/4
teslamate_1  |             (stdlib 3.16) gen_server.erl:750: :gen_server.handle_msg/6
teslamate_1  |             (stdlib 3.16) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  |     (elixir 1.12.3) lib/gen_server.ex:1024: GenServer.call/3
teslamate_1  |     (elixir 1.12.3) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
teslamate_1  |     (elixir 1.12.3) lib/task/supervised.ex:35: Task.Supervised.reply/5
teslamate_1  |     (stdlib 3.16) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  | Function: &:erlang.apply/2
teslamate_1  |     Args: [#Function<4.48828428/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [power: -11]]
teslamate_1  | 2021-10-06 17:35:14.327 [error] Task #PID<0.16521.0> started from #PID<0.16491.0> terminating
teslamate_1  | ** (stop) exited in: GenServer.call(TeslaMate.Mqtt.Publisher, {:publish, "teslamate/cars/1/frunk_open", "false", [retain: true, qos: 1]}, 10000)
teslamate_1  |     ** (EXIT) an exception was raised:
teslamate_1  |         ** (WithClauseError) no with clause matching: {:error, :timeout}
teslamate_1  |             (tortoise 0.10.0) lib/tortoise.ex:269: Tortoise.publish/4
teslamate_1  |             (teslamate 1.24.2) lib/teslamate/mqtt/publisher.ex:41: TeslaMate.Mqtt.Publisher.handle_call/3
teslamate_1  |             (stdlib 3.16) gen_server.erl:721: :gen_server.try_handle_call/4
teslamate_1  |             (stdlib 3.16) gen_server.erl:750: :gen_server.handle_msg/6
teslamate_1  |             (stdlib 3.16) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  |     (elixir 1.12.3) lib/gen_server.ex:1024: GenServer.call/3
teslamate_1  |     (elixir 1.12.3) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
teslamate_1  |     (elixir 1.12.3) lib/task/supervised.ex:35: Task.Supervised.reply/5
teslamate_1  |     (stdlib 3.16) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  | Function: &:erlang.apply/2
teslamate_1  |     Args: [#Function<4.48828428/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [frunk_open: false]]
teslamate_1  | 2021-10-06 17:35:14.332 [error] Task #PID<0.16517.0> started from #PID<0.16491.0> terminating
teslamate_1  | ** (stop) exited in: GenServer.call(TeslaMate.Mqtt.Publisher, {:publish, "teslamate/cars/1/latitude", "51.885783", [retain: true, qos: 1]}, 10000)
teslamate_1  |     ** (EXIT) an exception was raised:
teslamate_1  |         ** (WithClauseError) no with clause matching: {:error, :timeout}
teslamate_1  |             (tortoise 0.10.0) lib/tortoise.ex:269: Tortoise.publish/4
teslamate_1  |             (teslamate 1.24.2) lib/teslamate/mqtt/publisher.ex:41: TeslaMate.Mqtt.Publisher.handle_call/3
teslamate_1  |             (stdlib 3.16) gen_server.erl:721: :gen_server.try_handle_call/4
teslamate_1  |             (stdlib 3.16) gen_server.erl:750: :gen_server.handle_msg/6
teslamate_1  |             (stdlib 3.16) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  |     (elixir 1.12.3) lib/gen_server.ex:1024: GenServer.call/3
teslamate_1  |     (elixir 1.12.3) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
teslamate_1  |     (elixir 1.12.3) lib/task/supervised.ex:35: Task.Supervised.reply/5
teslamate_1  |     (stdlib 3.16) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  | Function: &:erlang.apply/2
teslamate_1  |     Args: [#Function<4.48828428/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [latitude: 51.885783]]
teslamate_1  | 2021-10-06 17:35:30.029 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-10-06 17:35:30.030 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-10-06 17:35:30.032 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-10-06 17:35:30.033 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-10-06 17:35:30.033 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-10-06 17:35:30.034 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-10-06 17:35:30.035 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-10-06 17:35:30.035 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-10-06 17:35:30.036 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-10-06 17:35:30.037 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-10-06 17:35:34.531 [error] GenServer TeslaMate.Mqtt.Publisher terminating
teslamate_1  | ** (WithClauseError) no with clause matching: {:error, :timeout}
teslamate_1  |     (tortoise 0.10.0) lib/tortoise.ex:269: Tortoise.publish/4
teslamate_1  |     (teslamate 1.24.2) lib/teslamate/mqtt/publisher.ex:41: TeslaMate.Mqtt.Publisher.handle_call/3
teslamate_1  |     (stdlib 3.16) gen_server.erl:721: :gen_server.try_handle_call/4
teslamate_1  |     (stdlib 3.16) gen_server.erl:750: :gen_server.handle_msg/6
teslamate_1  |     (stdlib 3.16) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  | Last message (from #PID<0.16528.0>): {:publish, "teslamate/cars/1/charger_power", "11", [retain: true, qos: 1]}
teslamate_1  | 2021-10-06 17:35:34.547 [error] Task #PID<0.16550.0> started from #PID<0.16525.0> terminating
teslamate_1  | ** (stop) exited in: GenServer.call(TeslaMate.Mqtt.Publisher, {:publish, "teslamate/cars/1/power", "-11", [retain: true, qos: 1]}, 10000)
teslamate_1  |     ** (EXIT) an exception was raised:
teslamate_1  |         ** (WithClauseError) no with clause matching: {:error, :timeout}
teslamate_1  |             (tortoise 0.10.0) lib/tortoise.ex:269: Tortoise.publish/4
teslamate_1  |             (teslamate 1.24.2) lib/teslamate/mqtt/publisher.ex:41: TeslaMate.Mqtt.Publisher.handle_call/3
teslamate_1  |             (stdlib 3.16) gen_server.erl:721: :gen_server.try_handle_call/4
teslamate_1  |             (stdlib 3.16) gen_server.erl:750: :gen_server.handle_msg/6
teslamate_1  |             (stdlib 3.16) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  |     (elixir 1.12.3) lib/gen_server.ex:1024: GenServer.call/3
teslamate_1  |     (elixir 1.12.3) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
teslamate_1  |     (elixir 1.12.3) lib/task/supervised.ex:35: Task.Supervised.reply/5
teslamate_1  |     (stdlib 3.16) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  | Function: &:erlang.apply/2
teslamate_1  |     Args: [#Function<4.48828428/1 in TeslaMate.Mqtt.PubSub.VehicleSubscriber.handle_info/2>, [power: -11]]
teslamate_1  | 2021-10-06 17:35:34.537 [error] Task #PID<0.16539.0> started from #PID<0.16525.0> terminating
teslamate_1  | ** (stop) exited in: GenServer.call(TeslaMate.Mqtt.Publisher, {:publish, "teslamate/cars/1/rated_battery_range_km", "175.58", [retain: true, qos: 1]}, 10000)
teslamate_1  |     ** (EXIT) an exception was raised:
teslamate_1  |         ** (WithClauseError) no with clause matching: {:error, :timeout}
teslamate_1  |             (tortoise 0.10.0) lib/tortoise.ex:269: Tortoise.publish/4
teslamate_1  |             (teslamate 1.24.2) lib/teslamate/mqtt/publisher.ex:41: TeslaMate.Mqtt.Publisher.handle_call/3
teslamate_1  |             (stdlib 3.16) gen_server.erl:721: :gen_server.try_handle_call/4
teslamate_1  |             (stdlib 3.16) gen_server.erl:750: :gen_server.handle_msg/6
teslamate_1  |             (stdlib 3.16) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  |     (elixir 1.12.3) lib/gen_server.ex:1024: GenServer.call/3
teslamate_1  |     (elixir 1.12.3) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
teslamate_1  |     (elixir 1.12.3) lib/task/supervised.ex:35: Task.Supervised.reply/5
```

### Type of installation

Docker

### Version

1.24.2

**Comment by ChrisHomewood** [comment]
In earlier versions of Teslamate I have found if MQTT is misconfigured then Teslamate tends to handle it not too gracefully, it would produce errors like the above and then not record the drive. Disabling MQTT in Teslamate solved the issue, as did eventually working out the correct config options!! I havn't tested with recent versions but it would be nice if Teslamate was more fault tollerant of MQTT failures or at least logged them better.

I also had problems with power supply on the Raspberry Pi that hosts Teslamate and it caused all manner of strange errors, it would interfere with the MQTT server generating loads of errors in the Mosquitto log, it would hang the Raspberry Pi if you did a particularly complex Grafana dashboard and also hang the Raspberry Pi at different times of the day. Took ages to work out this was the problem as the red power light was not flashing (which indicates insufficient voltage) so it was really marginal on the power supply being just ok, only became an issue at sustained high CPU load and SD card writing.

Anyway, my point is, a dodgy power supply can cause all manner of what look like software errors so there may not be a software problem here. In my case MQTT used to repeatedly disconnect clients which would then cause Teslamate to not record a drive or charge properly and other MQTT clients to also error. New bigger power supply fixed it.

**Comment by PieterAppronto** [comment]
Hi Chris,

You are right in your entire comment 😊
After buying the official Raspberry Pi charger all my issues were gone, I also disabled MQTT, but I don’t know whether this is really necessary, because it seemed that the charger fixed all my issues.

**Comment by MarcFinns** [comment]
I have this issue as well. Latest TeslaMate. I started noticing it a couple of days ago after updating to 2021.36.5.8. I had a 4 hours long international drive, and TeslaMate missed it almost completely. It only recorded the last 95km, as if the trip started in the middle of the highway. Looking a bit in the logs, I believe I found other instances of the behaviour. Tomorrow I will have another long drive and will check. Let me know what info would be useful to better understand the issue. Thanks.
